velox/expression/tests/ExpressionFuzzerVerifier.h

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "velox/core/ITypedExpr.h"
+#include "velox/core/QueryCtx.h"
+#include "velox/expression/Expr.h"
+#include "velox/expression/tests/ExpressionFuzzer.h"
+#include "velox/expression/tests/ExpressionVerifier.h"
+#include "velox/expression/tests/utils/FuzzerToolkit.h"
+#include "velox/functions/FunctionRegistry.h"
+#include "velox/vector/fuzzer/VectorFuzzer.h"
+#include "velox/vector/tests/utils/VectorMaker.h"
+
+DECLARE_int32(velox_fuzzer_max_level_of_nesting);
+
+namespace facebook::velox::test {
+
+// A tool utilizes ExpressionFuzzer, VectorFuzzer and ExpressionVerfier to
+// generate random expressions and verify the correctness of the results.
+class ExpressionFuzzerVerifier {
+ public:
+  ExpressionFuzzerVerifier(
```

Comment:
I think possibly we should also pass in ExpressionFuzzer, vector fuzzer and expression verifier to take full advantage of this refactoring.

Reply:
I replied to this in a comment above, I am doing that in a different PR. is it ok if we move this discussion there https://github.com/facebookincubator/velox/pull/7634